File: nodejs/examples/basic_usage.js
const { MandiriQrisClient } = require('../src');

// Initialize client
const client = new MandiriQrisClient({
    clientId: process.env.MANDIRI_CLIENT_ID || 'your_client_id',
    clientSecret: process.env.MANDIRI_CLIENT_SECRET || 'your_client_secret',
    baseUrl: process.env.MANDIRI_BASE_URL || 'https://api.mandiri.co.id',
    sandbox: true
});

async function main() {
    try {
        // Generate QR Code
        console.log('Generating QR Code...');
        const qr = await client.generateQR({
            amount: 100000,
            merchantId: 'MERCHANT123',
            terminalId: 'TERM001',
            customerName: 'John Doe',
            customerPhone: '081234567890'
        });
        
        console.log('QR Generated Successfully!');
        console.log('Transaction ID:', qr.transaction_id);
        console.log('QR String:', qr.qr_string);
        console.log('');
        
        // Check payment status
        console.log('Checking payment status...');
        const status = await client.checkPaymentStatus(qr.transaction_id);
        console.log('Status:', status.status);
        console.log('Amount:', status.amount);
        console.log('');
        
        // Poll payment status (optional)
        console.log('Polling payment status (will wait up to 5 minutes)...');
        const finalStatus = await client.pollPaymentStatus(qr.transaction_id, 60, 5);
        console.log('Final Status:', finalStatus.status);
        
    } catch (error) {
        console.error('Error:', error.message);
    }
}

main();

File: python/examples/basic_usage.py
"""
Basic usage example for Mandiri QRIS Python SDK
"""

import os
from mandiri_qris import MandiriQrisClient

# Initialize client
client = MandiriQrisClient(
    client_id=os.getenv('MANDIRI_CLIENT_ID', 'your_client_id'),
    client_secret=os.getenv('MANDIRI_CLIENT_SECRET', 'your_client_secret'),
    base_url=os.getenv('MANDIRI_BASE_URL', 'https://api.mandiri.co.id'),
    sandbox=True
)

try:
    # Generate QR Code
    print("Generating QR Code...")
    qr = client.generate_qr(
        amount=100000,
        merchant_id='MERCHANT123',
        terminal_id='TERM001',
        customer_name='John Doe',
        customer_phone='081234567890'
    )
    
    print("QR Generated Successfully!")
    print(f"Transaction ID: {qr['transaction_id']}")
    print(f"QR String: {qr['qr_string']}\n")
    
    # Check payment status
    print("Checking payment status...")
    status = client.check_payment_status(qr['transaction_id'])
    print(f"Status: {status['status']}")
    print(f"Amount: {status['amount']}\n")
    
    # Poll payment status (optional)
    print("Polling payment status (will wait up to 5 minutes)...")
    final_status = client.poll_payment_status(qr['transaction_id'], 60, 5)
    print(f"Final Status: {final_status['status']}")
    
except Exception as e:
    print(f"Error: {e}")
